agreementbot/agreementbot.go

```diff
@@ -1119,6 +1122,51 @@ func (w *AgreementBotWorker) syncOnInit() error {
 						glog.V(3).Infof(AWlogString(fmt.Sprintf("added agreement %v to policy agreement counter.", ag.CurrentAgreementId)))
 					}
 
+					// After checking the pol userdev/bp_netspeed, add it in to a map. In Each for loop which iterate the agreements, checking if current policy inside agreement has been handled or not
+					// re-evaluate the agreement
+					if pol != nil && !bPolicyCheckingMap[pol.Header.Name] {
+						glog.V(3).Infof(AWlogString(fmt.Sprintf("checking policy against exchange for agreement %v.", ag.CurrentAgreementId)))
+						// policyName example:
+						//     - userdev/bp_netspeed
+						if exchPols, err := exchange.GetBusinessPolicies(w, exchange.GetOrg(pol.Header.Name), exchange.GetId(pol.Header.Name)); err != nil {
+							glog.Errorf(AWlogString(fmt.Sprintf("error getting business policies from exchange for org: %v, policy name: %v error: %v", ag.Org, pol.Header.Name, err)))
+						} else if len(exchPols) == 0 {
+							glog.V(3).Infof(AWlogString(fmt.Sprintf("business policy %v from agreement %v is not found from exchange.", pol.Header.Name, ag.CurrentAgreementId)))
+							// Need to cancel the agreement
+							policyDeletedMsg := events.NewPolicyDeletedMessage(events.DELETED_POLICY, "", pol.Header.Name, exchange.GetOrg(pol.Header.Name), ag.Policy)
+							pdCmd := NewPolicyDeletedCommand(*policyDeletedMsg)
+							// Queue the command to the relevant protocol handler for further processing.
+							if w.consumerPH.Get(agp).AcceptCommand(pdCmd) {
+								w.consumerPH.Get(agp).HandlePolicyDeleted(pdCmd, w.consumerPH.Get(agp))
+							}
+						} else {
+							for polId, exchPol := range exchPols {
+								bPol := exchPol.GetBusinessPolicy()
+								if exPolicy, err := bPol.GenPolicyFromBusinessPolicy(polId); err != nil {
+									glog.Errorf(AWlogString(fmt.Sprintf("error generating internal business policies for org: %v, policy name: %v from %v, error: %v", ag.Org, pol.Header.Name, exchPol.String(), err)))
+								} else if exPolicy == nil {
+									glog.Errorf(AWlogString(fmt.Sprintf("the generated internal business policies is nil for org: %v, policy name: %v from %v", ag.Org, pol.Header.Name, exchPol.String())))
+								} else if exPolicyString, err := policy.MarshalPolicy(exPolicy); err != nil {
+									glog.Errorf(fmt.Sprintf("Error trying to marshal internal business policy %v error: %v", exPolicy, err))
+								} else {
+									// If business policy has been changed during a restart, handle it
+									glog.V(3).Infof(AWlogString(fmt.Sprintf("policy(%v) in agreement %v has been changed", pol.Header.Name, ag.CurrentAgreementId)))
```

Comment:
I don't think this log message is correct... At this point, we don't know that the policy has been changed... we just want the agbot to re-evaluate the policy just in case it changed.